File: backend/src/main/kotlin/dtos/YearTrendDto.kt
package com.xavierclavel.dtos

import com.xavierclavel.utils.BigDecimalSerializer
import kotlinx.serialization.Serializable
import java.math.BigDecimal


@Serializable
data class YearTrendDto(
    val year: Int,
    val month: Int? = null,
    @Serializable(with = BigDecimalSerializer::class)
    val total: BigDecimal,
    @Serializable(with = BigDecimalSerializer::class)
    val median: BigDecimal,
    @Serializable(with = BigDecimalSerializer::class)
    val average: BigDecimal,
) {
    constructor(year: Int, total: BigDecimal, median: BigDecimal, average: BigDecimal) : this(
        year = year,
        month = null,
        total = total,
        median = median,
        average = average,
    )
}

File: frontend/src/types/YearTrendDto.ts
import SubcategoryOut from "@/src/types/SubcategoryOut";
import CategorySummary from "@/src/types/CategorySummary";

export default class YearTrendDto{
    public constructor(
        public year: number | null,
        public month: number | null,
        public total: string,
        public average: string,
        public median: string,
        ) {}
}